File: counting2.py
def FindMax(arr):
    n = len(arr)
    max = arr[0]
    for i in range(1, n):
        if arr[i] > max:
            max = arr[i]
    return max
def main():

    arr = [9,8,7,6,5,5,4,4,3,2,1,3,9]
    n = len(arr)

    print(arr)

    max = FindMax(arr)

    counts = [ 0 ] * (max + 1)

    for i in range(0, n):

        counts[ arr[i] ] += 1

    pos = 0

    for i in range(0, max+1):

        for j in range(0, counts[i]):

            arr[pos] = i

            pos +=1

    print(arr)

main()

File: counting2.php
<?php
/*
Author:
Adrian Statescu <mergesortv@gmail.com>

Description:
Counting Sort Algorithm.

Analysis Complexity:
Worst Case Time: O(n)
Average Case Time: O(n)
Best Case Time: O(n)
Space: O(n)

*/

function GetMax($arr, $n) {

   $max = $arr[0];

   for($i = 1; $i < $n; $i++) {
       if($arr[$i] > $max) {
         $max = $arr[$i];
       }
   }
   return $max;
}
$arr = array(4,8,7,6,5,15,9,8,1);

$n = count($arr);
$max = GetMax($arr,$n);

$counts = array();

for($i = 0; $i <= $max; $i++) {
  $counts[$i] = 0;
}

echo"Input:";
echo"<pre>";
print_r($arr);
echo"</pre>";


$max = GetMax($arr,$n);

for($i = 0; $i < $n; $i++) {
    $counts[$arr[$i]] += 1;
}

$pos = 0;
for($i = 0; $i <= $max;$i++) {
    for($j = 0; $j < $counts[$i]; $j++) {
        $arr[$pos] = $i;
        $pos = $pos + 1;
    }
}

echo"Output:";
echo"<pre>";
print_r($arr);
echo"</pre>";

?>
